homeassistant/components/boiler/__init__.py

```diff
@@ -0,0 +1,496 @@
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_AWAY_MODE, ATTR_AWAY_MODE)
+            return
+
+        for boiler in target_boiler:
+            if away_mode:
+                boiler.turn_away_mode_on()
+            else:
+                boiler.turn_away_mode_off()
+
+            if boiler.should_poll:
+                boiler.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_AWAY_MODE, away_mode_set_service,
+        descriptions.get(SERVICE_SET_AWAY_MODE),
+        schema=SET_AWAY_MODE_SCHEMA)
+
+    def guest_mode_set_service(service):
+        """Set guest mode on target boiler devices."""
+        target_boiler = component.extract_from_service(service)
+
+        guest_mode = service.data.get(ATTR_GUEST_MODE)
+        total_guests = service.data.get(ATTR_TOTAL_GUESTS)
+
+        if guest_mode is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_GUEST_MODE, ATTR_GUEST_MODE)
+            return
+
+        for boiler in target_boiler:
+            if guest_mode:
+                boiler.turn_guest_mode_on(total_guests)
+            else:
+                # Assuming no needs to set days to zero, thus no kwargs
+                boiler.turn_guest_mode_off()
+
+            if boiler.should_poll:
+                boiler.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_GUEST_MODE, guest_mode_set_service,
+        descriptions.get(SERVICE_SET_GUEST_MODE),
+        schema=SET_GUEST_MODE_SCHEMA)
+
+    def holiday_mode_set_service(service):
+        """Set holiday mode on target boiler devices."""
+        target_boiler = component.extract_from_service(service)
+
+        holiday_mode = service.data.get(ATTR_HOLIDAY_MODE)
+        holiday_duration = service.data.get(ATTR_HOLIDAY_DURATION)
+
+        if holiday_mode is None:
+            _LOGGER.error(
+                "Received call to %s without attribute %s",
+                SERVICE_SET_HOLIDAY_MODE, ATTR_HOLIDAY_MODE)
+            return
+
+        for boiler in target_boiler:
+            if holiday_mode:
+                boiler.turn_holiday_mode_on(holiday_duration)
+            else:
+                # Assuming no needs to set days to zero, thus no kwargs
+                boiler.turn_holiday_mode_off()
+
+            if boiler.should_poll:
+                boiler.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_HOLIDAY_MODE, holiday_mode_set_service,
+        descriptions.get(SERVICE_SET_HOLIDAY_MODE),
+        schema=SET_HOLIDAY_MODE_SCHEMA)
+
+    def temperature_set_service(service):
+        """Set temperatures on the target climate devices."""
+        target_boiler = component.extract_from_service(service)
+
+        for boiler in target_boiler:
+            kwargs = {}
+            for value, temp in service.data.items():
```

Comment:
entity_id can also be part of `items`. 

Also it seems that you're doing this for loop to iterate over 2 attributes, would it make more sense for readability to just write it out?